### DISCUSSION TITLE
Parameters for Query

### DISCUSSION THREAD

**Discussion by sreiach** [TEAMSchools/powerschool#29] [Q&A] [answered]
I'm missing something.  I'm able to connect to my PS and pull basic table data.  I'm now trying to pull data just on one student.  However, no matter what I do - it always returns ALL students.

students_table.query(student_number=1234) returns all students (active and inactive).
students_table.query(dcid=1234) returns all students..

param = {'student_number' : '1234' }
student_table.query(body = param) returns all students. 

How should I be doing it?

**Comment by cbini** [comment, marked as answer]
Hey! good question--the `README` was wrong. The correct parameter to pass is `pk`, e.g. `students_table.query(pk=1234)`.

I think I changed it a while back because the primary key turned out to not necessarily be `dcid` for all tables.
